coderd/database/dbauthz/dbauthz.go

```diff
@@ -158,34 +158,49 @@ func ActorFromContext(ctx context.Context) (rbac.Subject, bool) {
 }
 
 var (
-	subjectProvisionerd = rbac.Subject{
-		FriendlyName: "Provisioner Daemon",
-		ID:           uuid.Nil.String(),
-		Roles: rbac.Roles([]rbac.Role{
-			{
-				Identifier:  rbac.RoleIdentifier{Name: "provisionerd"},
-				DisplayName: "Provisioner Daemon",
-				Site: rbac.Permissions(map[string][]policy.Action{
-					// TODO: Add ProvisionerJob resource type.
-					rbac.ResourceFile.Type:     {policy.ActionRead},
-					rbac.ResourceSystem.Type:   {policy.WildcardSymbol},
-					rbac.ResourceTemplate.Type: {policy.ActionRead, policy.ActionUpdate},
-					// Unsure why provisionerd needs update and read personal
-					rbac.ResourceUser.Type:             {policy.ActionRead, policy.ActionReadPersonal, policy.ActionUpdatePersonal},
-					rbac.ResourceWorkspaceDormant.Type: {policy.ActionDelete, policy.ActionRead, policy.ActionUpdate, policy.ActionWorkspaceStop},
-					rbac.ResourceWorkspace.Type:        {policy.ActionDelete, policy.ActionRead, policy.ActionUpdate, policy.ActionWorkspaceStart, policy.ActionWorkspaceStop},
-					rbac.ResourceApiKey.Type:           {policy.WildcardSymbol},
-					// When org scoped provisioner credentials are implemented,
-					// this can be reduced to read a specific org.
+	subjectProvisionerd = func(orgID uuid.UUID) rbac.Subject {
+		sitePermissions := map[string][]policy.Action{
+			// TODO: Add ProvisionerJob resource type.
+			rbac.ResourceFile.Type:     {policy.ActionRead},
+			rbac.ResourceSystem.Type:   {policy.WildcardSymbol},
+			rbac.ResourceTemplate.Type: {policy.ActionRead, policy.ActionUpdate},
+			// Unsure why provisionerd needs update and read personal
+			rbac.ResourceUser.Type:             {policy.ActionRead, policy.ActionReadPersonal, policy.ActionUpdatePersonal},
+			rbac.ResourceWorkspaceDormant.Type: {policy.ActionDelete, policy.ActionRead, policy.ActionUpdate, policy.ActionWorkspaceStop},
+			rbac.ResourceWorkspace.Type:        {policy.ActionDelete, policy.ActionRead, policy.ActionUpdate, policy.ActionWorkspaceStart, policy.ActionWorkspaceStop},
+			rbac.ResourceApiKey.Type:           {policy.WildcardSymbol},
+			// When org scoped provisioner credentials are implemented,
+			// this can be reduced to read a specific org.
+			rbac.ResourceOrganization.Type: {policy.ActionRead},
+			rbac.ResourceGroup.Type:        {policy.ActionRead},
+		}
+		orgPermissions := map[string][]rbac.Permission{}
+
+		if orgID != uuid.Nil {
+			// replace site wide org permissions with org scoped permissions
+			delete(sitePermissions, rbac.ResourceOrganization.Type)
+			orgPermissions = map[string][]rbac.Permission{
+				orgID.String(): rbac.Permissions(map[string][]policy.Action{
 					rbac.ResourceOrganization.Type: {policy.ActionRead},
-					rbac.ResourceGroup.Type:        {policy.ActionRead},
 				}),
-				Org:  map[string][]rbac.Permission{},
-				User: []rbac.Permission{},
-			},
-		}),
-		Scope: rbac.ScopeAll,
-	}.WithCachedASTValue()
```

Comment:
This only replaces the `rbac.ResourceOrganization.Type: {policy.ActionRead},` perm. We should probably scope everything to the org. All the perms in the site should be moved.

If there is perm issues for not granting site wide, we will have to address that in another PR.

Reply:
We talked and I am not including any rbac changes in this PR